src/OrchardCore/OrchardCore.Environment.Shell.Abstractions/Builders/ShellContext.cs

```diff
@@ -124,12 +124,19 @@ internal class ServiceScopeWrapper : IServiceScope
             private readonly IServiceProvider _existingServices;
             private readonly HttpContext _httpContext;
 
-            public ServiceScopeWrapper(IServiceScope serviceScope)
+            public ServiceScopeWrapper(IServiceScope serviceScope, bool newHttpContext = false)
             {
                 ServiceProvider = serviceScope.ServiceProvider;
-
                 _serviceScope = serviceScope;
-                _httpContext = ServiceProvider.GetRequiredService<IHttpContextAccessor>().HttpContext;
+
+                var httpContextAccessor = ServiceProvider.GetRequiredService<IHttpContextAccessor>();
+
+                if (newHttpContext)
+                {
+                    httpContextAccessor.HttpContext = new DefaultHttpContext();
```

Comment:
Aren't there any properties we want to copy for this new scope?

Reply:
Yes, i was thinking about this, as we did in O1 e.g to simulate a request with a fake path but tenant aware ... After a quick look it seems that now more properties are automatically created if not defined.

I need to see it more in details, **maybe in another PR**, as you want.

Note: Just [saw this](https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNetCore.Owin/OwinExtensions.cs#L138), we would not be the only ones to use the `DefaultHttpContext`.

Reply:
This still feels weird, IMHO. Won't that replace the context of the request that initiates the tenant? If the background task service does that in the middle of the request, that may be terrible.

Reply:
I'm never sure of nothing but i tried it, the reference of the object is flowed up among threads which are sharing the same object reference (so not thread safe if the object is not), but if in a child thread we assign a new object, as i have checked, it seems that it doesn't affect the parent thread reference to the original object which still exists. Here it seems that it works as the `LogicalCallContext` which only make shallow copy (not cloned in a new object), but if we assign a new value it seems to be ok.

But you're right, it needs to be checked again with caution.

Reply:
What do you think of suppressing the execution context in the background task service? Just to be sure (this code makes me nervous :cold_sweat:)

Reply:
No problem, i will re-think about this. Anyway i will not propose it if you are not sure yourself ;)

Then i think that what i said would not be true if tomorrow we don't execute bgTask in a thread that we explicitly start as with the timer, and we would need to be sure that a timer task never execute in the same thread of the one which started it.